consensus/safety-rules/src/process_client_wrapper.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{safety_rules_manager, ConsensusState, Error, SafetyRulesManager, TSafetyRules};
+use consensus_types::{
+    block::Block,
+    block_data::BlockData,
+    common::{Payload, Round},
+    quorum_cert::QuorumCert,
+    timeout::Timeout,
+    vote::Vote,
+    vote_proposal::VoteProposal,
+};
+use libra_config::{
+    config::{ConsensusType, NodeConfig, RemoteService, SafetyRulesBackend, SafetyRulesService},
+    utils,
+};
+use libra_types::crypto_proxies::{Signature, ValidatorSigner};
+use std::{
+    any::TypeId,
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+};
+
+/// This container exists only so that we can kill the spawned process after testing is complete.
+/// Otherwise the process will be killed at the end of the safety_rules function and the test will
+/// fail.
+pub struct ProcessClientWrapper<T> {
```

Comment:
this file belongs to `src/tests/` instead of `/src/`?

Reply:
I put the test_utils in src. If we put it in src/tests/ then we have to make the test module public. I don't know if there is a proper Rust way of doing this...